lib/workers/global/config/parse/index.ts

```diff
@@ -109,6 +110,16 @@ export async function parseConfigs(
     config.onboardingNoDeps = 'enabled';
   }
 
+  // do not add these secrets to repoSecrets and,
+  //  do not delete the secrets object after applying on global config as it needs to be re-used for repo config
+  if (is.nonEmptyObject(config.secrets)) {
+    config = applySecretsToConfig(config, undefined, false);
```

Comment:
If this is applied, is https://github.com/renovatebot/renovate/blob/68a66a05a98ae2ed77ec2f1e203fac553383b081/lib/workers/global/initialize.ts#L64 also still required?

Reply:
It should not be. But, even in the `globalInitialize` function `setGlobalHostRules` has been called twice, which mean that some host rules could get added during `initPlatform` call. Hence, better to leave it there,